src/components/Connection.tsx

```diff
@@ -610,25 +612,46 @@ const Connection: React.FC<ConnectionProps> = ({
   const saveData = async () => {
     try {
       const allData = await getAllDataFromIndexedDB();
-
+  
       if (allData.length === 0) {
         toast.error("No data available to download.");
         return;
       }
-
-      // Ensure data is in the correct format
-      const formattedData = allData.map((item) => ({
-        timestamp: item.timestamp,
-        channel_1: item.channel_1,
-        channel_2: item.channel_2,
-        channel_3: item.channel_3,
-        channel_4: item.channel_4,
-      }));
-
+  
+      // Log the raw data from IndexedDB to ensure the channels are correctly stored
+      console.log('Raw data from IndexedDB:', allData);
```

Comment:
We should not keep any debug console log in the upstream code, please remove this if this is only for debugging purposes.

Reply:
fixed this issue.